src/riscv/lib/src/state_backend/proof_backend/proof/deserialiser.rs

```diff
@@ -185,6 +188,26 @@ pub trait Suspended {
     where
         Self::Output: 'static;
 }
+
+/// Trait for common deserialisation functionality abstracting over the deserialiser type.
+pub trait RunDeserialiser: Deserialiser + Sized {
+    /// Data type consumed by the deserialiser.
+    type Data;
```

Comment:
Inputs are often more suited to be represented as trait parameters. For example, in this case, the `StreamDeserialiser` could work on any `T: AsRef<[u8]>`.